HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,343 @@
+
+        /*************************************************************************************************************
+         *      Direction Zivid to Robot ( Rotation Matrix --> any format)
+         *************************************************************************************************************/
+        const Representations retVals = transformFromZividToRobot(rotationMatrix);
+
+        /*************************************************************************************************************
+         *      Direction robot to Zivid (any format --> Rotation Matrix)
+         *************************************************************************************************************/
+        transformFromRobotToZivid(rotationMatrix, retVals);
+    }
+
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << e.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+cv::Mat eigenToCv(const Eigen::MatrixXd &eigenMat)
+{
+    cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+    for(int i = 0; i < cvMat.rows; i++)
+    {
+        for(int j = 0; j < cvMat.cols; j++)
+        {
+            cvMat.at<double>(i, j) = eigenMat(i, j);
+        }
+    }
+
+    return cvMat;
+}
+
+Eigen::MatrixXd cvToEigen(const cv::Mat &cvMat)
+{
+    if(cvMat.dims > 2)
+    {
+        throw "Invalid matrix dimensions. Expected 2D.";
+    }
+
+    Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+    for(int i = 0; i < cvMat.rows; i++)
+    {
+        for(int j = 0; j < cvMat.cols; j++)
+        {
+            eigenMat(i, j) = cvMat.at<double>(i, j);
+        }
+    }
+
+    return eigenMat;
+}
+
+Representations transformFromZividToRobot(const Eigen::Matrix3d &rotationMatrix)
+{
+    Representations retVals;
+    // Rotation Matrix in Transformation Matrix to Axis-angle
+    std::cout << "\nConverting rotation matrix to axis-angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle = Eigen::AngleAxisd(rotationMatrix);
+    retVals.angleAxis = reorderAxisAngle(axisAngle);
+    std::cout << "\nAxis-angle:\n" << retVals.angleAxis << std::endl;
+
+    // Axis-angle to Rotation Vector
+    std::cout << "\nConverting axis-angle to rotation vector" << std::endl;
+    if(!axisAngle.axis().isUnitary())
+    {
+        std::cout << "Note! Axis in axis-angle pair is not unitary" << std::endl;
+    }
+    retVals.rotationVector = axisAngle.angle() * axisAngle.axis();
+    std::cout << "\nRotationVector:\n" << retVals.rotationVector << std::endl;
+
+    // Rotation Matrix to Quaternion
+    std::cout << "\nConverting rotation matrix to quaternion" << std::endl;
+    const Eigen::Quaterniond quaternion = Eigen::Quaterniond(rotationMatrix);
+    retVals.quaternion = quaternion.coeffs();
+    std::cout << "\nQuaternion:\n" << retVals.quaternion << std::endl;
+
+    // Rotation Matrix to Roll-Pitch-Yaw
```

Comment:
Suggestion: If you simplify `RollPitchYaw` as I suggest below, you can simplify the initialization of `retVals.rotations` quite a bit by removing this initial initialization and do everything in the for loop:
```c++
    for(int i = 0; i < nofRotConv; i++)
    {
        const RotationConvention convention{ static_cast<RotationConvention>(i) };
        std::cout << "\nConverting from rotation matrix to roll-pitch-yaw angles (" << toString(convention) << ")"
                  << std::endl;
        retVals.rotations[i] = { static_cast<RotationConvention>(i), rotmat2rpy(rotationMatrix, convention) };
        std::cout << "\nRollPitchYaw" << toString(convention) << ":\n"
                  << retVals.rotations[i].rollPitchYaw << std::endl;
    }

```